app/serializers/serialize_submission.rb

```diff
@@ -15,9 +15,12 @@ def call
           submission.solution.uuid,
           submission_id: submission.uuid,
           auth_token: user.auth_tokens.first.to_s
+        ),
+        test_run: Exercism::Routes.api_submission_test_run_url(
+          submission.uuid,
```

Comment:
This is fine, because this is at serialization. Here you can build the URL and then never again.